homeassistant/components/wmspro/diagnostics.py

```diff
@@ -0,0 +1,23 @@
+"""Diagnostics support for WMS WebControl pro API integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    diag = {
+        "control": entry.runtime_data.__dict__,
+        "dests": {k: v.__dict__ for k, v in entry.runtime_data.dests.items()},
+        "rooms": {k: v.__dict__ for k, v in entry.runtime_data.rooms.items()},
+        "scenes": {k: v.__dict__ for k, v in entry.runtime_data.scenes.items()},
```

Comment:
Do we have a nice way to show the data? `__dict__` isn't very beautiful

Reply:
I guess I should probably add a proper raw method to pywmspro and then use that here.

Reply:
I think that's the best way yes. It will also help with the snapshot tests so we can have a full view of the data used

Reply:
Done.